app/controllers/SndConHavePhoneController.scala

```diff
@@ -42,46 +44,35 @@ class SndConHavePhoneController @Inject() (
   standardActionSets: StandardActionSets,
   formProvider: SndConHavePhoneFormProvider,
   val controllerComponents: MessagesControllerComponents,
-  renderer: Renderer
+  view: SndConHavePhoneView
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
-    with NunjucksSupport {
+    with ContactHelper {
 
   private val form = formProvider()
 
-  private def render(mode: Mode, form: Form[Boolean], name: String)(implicit request: DataRequest[AnyContent]): Future[Html] = {
-    val data = Json.obj(
-      "form"   -> form,
-      "name"   -> name,
-      "action" -> routes.SndConHavePhoneController.onSubmit(mode).url,
-      "radios" -> Radios.yesNo(form("value"))
-    )
-    renderer.render("sndConHavePhone.njk", data)
-  }
+  def onPageLoad(mode: Mode): Action[AnyContent] = standardActionSets.identifiedUserWithData() {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(SndConHavePhonePage) match {
+        case None        => form
+        case Some(value) => form.fill(value)
+      }
 
-  def onPageLoad(mode: Mode): Action[AnyContent] =
-    standardActionSets.identifiedUserWithDependantAnswer(SndContactNamePage).async {
-      implicit request =>
-        val preparedForm = request.userAnswers.get(SndConHavePhonePage) match {
-          case None        => form
-          case Some(value) => form.fill(value)
-        }
-        render(mode, preparedForm, request.userAnswers.get(SndContactNamePage).get).map(Ok(_))
-    }
+      Ok(view(preparedForm, getSecondContactName(request.userAnswers), mode))
+  }
 
```

Comment:
in old code identifiedUserWithDependantAnswer is used , while for the latest one we are using identifiedUserWithData() , just check once if its ok.

Reply:
rest looks ok.